psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -481,7 +484,13 @@ def scf_iterate(self, e_conv=None, d_conv=None):
 
         # Call any postiteration callbacks
         if not ((self.iteration_ == 0) and self.sad_) and _converged(Ediff, Dnorm, e_conv=e_conv, d_conv=d_conv):
-            break
+            if early_screening:
+                early_screening = False
+                self.jk().set_early_screening(False)
```

Comment:
is the idea that E_CONV and D_CONV values in the options object are going to be changing between the early_screening T and Fa stages? otherwise, I don't think I'm following the iteration logic.

Reply:
Not quite; `E_CONV` and `D_CONV` don't change. The idea is that JK algorithms have various parameters and thresholds that control accuracy (think `INTS_TOLERANCE`). Instead of using a tight `INTS_TOLERANCE` throughout a whole SCF calculation, you could get away with using a loose `INTS_TOLERANCE` in the earlier iterations, and then tightening the parameter in the final iterations.

`early_screening_` is a boolean attached to the `JK` class that tells the class whether or not it's allowed to get away with such "looser" cutoffs. The iteration logic works like this:  
 - If `early_screening_` is off, SCF runs to convergence as it always has.
  - If `early_screening_` is on, SCF runs to convergence, then switches `early_convergence_` to off. We then resume the SCF iterations because the switch from on to off may slightly shift the location of the SCF solution and require another iteration or two.

I've set `early_screening_` to false in all existing `JK` classes, so it doesn't currently affect any behavior. In semi-numerical exchange, the cost is proportional to the size of the integration grid. You can get a big performance boost by using a small grid until SCF convergence, and then switch to a large grid for a final iteration.

Reply:
Ok, got it, I think. `INTS_TOLERANCE` or similar plays the role of hidden variable in the SCF iterations whose value is influenced by `early_screening_`. I couldn't figure how querying early_screening before iterations and resetting it after iterations was going to work. Thanks for the explanation.

Reply:
Exactly! I'm open to suggestions if you can think of a more immediately understandable variable name than `early_screening_`

Reply:
`early_screening_` is fine by me. It was only the mechanism of `early_screening_` that confused me, and that'll be traceable once it appears in a JK.